app.rb

```diff
@@ -779,11 +779,15 @@ def render_template(name)
       error 400,
         :error => _("The nic_max parameter must be an integer not starting with 0")
 
-    params['http_port'].nil? and request.secure? and
-      error 400,
-        :error => _("The `http_port` argument must be supplied for bootstrap generation on a secure port")
-
-    @bootstrap_port = params['http_port'].nil? ? request.port.to_s : params['http_port']
+    params["http_port"].nil? or (params["http_port"] =~ /\A[1-9][0-9]?[0-9]?[0-9]?[0-9]?\Z/ and
```

Comment:
I have to google to find out what operator precedence in ruby with "and" / "or" is, which is a big hit to readability. perhaps clean up with parens or intermediate variables? this is really hard to read and parse.